### PR TITLE
Show unit in numeric state trigger and condition

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

While working on #13628 I thought about the corresponding logic for the numeric state trigger and condition and the unit of measurement was completely ignored there so far. This PR now adds it in (as long as no attribute is selected, since then it makes no sense).

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-selector/ha-selector-number.ts [hunk 1]
```diff
@@ -29,13 +29,17 @@ export class HaNumberSelector extends LitElement {
   protected render() {
     const isBox = this.selector.number.mode === "box";
 
+    const fullLabel = `${this.label}${
```
Comment (spacegaier) [inline src/components/ha-selector/ha-selector-number.ts:32]: Worth to put into a memoizeOne()?

Comment (bramkragten) [reply src/components/ha-selector/ha-selector-number.ts:32]: Don't put this in the label, we have the suffix for this

**Comment by spacegaier** [comment]
> this is already in the suffix right? why also in the label?

Because the suffix only gets shown once there is a value in the field, before that you only have the label where seeing the unit I think is also already helpful.

**Comment by bramkragten** [comment]
> > this is already in the suffix right? why also in the label?
> 
> Because the suffix only gets shown once there is a value in the field, before that you only have the label where seeing the unit I think is also already helpful.

It is shown as soon as you have focus on the field, so when you are gonna put a value in, I guess that makes sense?

**Comment by bramkragten** [comment]
It should definitely not be in `ha-selector`, if anywhere (I don't think it should so personally) this label logic should be in the consumer of the selector

**Pull request closed without merging**